### ISSUE TITLE
Tender Module

### ISSUE THREAD

**Issue by rdmordido** [leisuretexpainting/leisuretexpainting#5]
This is the main module for creating and managing tenders.

**Comment by rdmordido** [comment]
@leisuretexpainting 

I am now starting to build the tender module and all its features.
I have some questions here:
1. What does ABN stands for?
2. How many contractor has a contact can be assigned to? (one or many)
3. What is the importance of a contact when creating a new tender? Do we really  need to input the contact information during this stage?
4. How many grade does a contact belongs to?
5. Who are Sales Representative and what is the importance of it when encoding a new job?

**Comment by rdmordido** [comment]
attaching preview of tender setup page (temporary)
![create_tender](https://cloud.githubusercontent.com/assets/1768593/5676629/1d31c5d6-9814-11e4-80bc-1625b2c5e5e6.JPG)

**Comment by leisuretexpainting** [comment]
What does ABN stands for? - Australian Business Number - each business in Australia must have one and we use it for taxation purposes. The customer's abn is important.

How many contractor has a contact can be assigned to? (one or many) A contractor can have many different contacts. A contact though can change between contractors over time. For example Contact A can work for Contractor A for a few years but then change jobs to work for Contractor B. We often will stay in touch with Contact A when they move to Contractor B and continue to price for them. They will keep their grading etc until we change it depending on what work we receive from them.

What is the importance of a contact when creating a new tender? Do we really need to input the contact information during this stage? Very important. We base the importance of the tender on who the contact is. Also throughout the the tender stage our estimators will need to know who to contact with any queries they have. It is also our protocol that after a tender is received the Senior Estimator gives a follow up call to the Contact to let them know we will be tendering on the project / ask any questions etc. (This is why in the job file I would like for them to be able to note any correspondence in regards to the tender).

How many grade does a contact belongs to? A contact will be graded A, B or C. Once they have been graded that will determine the importance of the tender. A grade can be changed at any time.

Who are Sales Representative and what is the importance of it when encoding a new job? Our Sales Representatives will be the user who has brought the tender to us in the first place. We have 2 sales reps and they each liaise with different contacts and clients. If an estimator needs further information on a project they can check who the salesman was to confirm with them. Also we will be using this note of the sales rep later to see who is bringing in which projects, the dollar value assigned to each project etc.

Does this help? :)

**Comment by rdmordido** [comment]
@leisuretexpainting ,

Thanks, those are useful information.
In addition to my inquiries above,

Based on our user types as per the last task from Issue #4,

Which group has the capability to create or initialize the tender process?
Which group does the Sales Representative belongs to?
Can I have a sample of our Organization Chart as reference?

Administrator
Sales
Operations
Senior Estimator
Estimator
Supervisor
Staff
Contractor

**Comment by rdmordido** [comment]
@leisuretexpainting 

I need to have the final the list of stages / status that we will be using to track each tender during the tender process.
From the old database, I saw these:

Awaiting take off material
Awaiting Quotation
Quotation sent

**Comment by rdmordido** [comment]
@leisuretexpainting 

After a tender has been created / initialized What will be the next process?
How does a Tender and a Quotation works?

I am trying to summarize the relationship between a new tender and the quotation process.
Please correct me if I am wrong, Quotation comes after a tender has been initialized wherein there will be specific computations per items such as the item, number of units, rate per item, and the total amount etc..
The total amount will then be automatically computed and encoded under the main tender table which also includes the contractor id,contact id, sales representative's id, and the job id.

**Comment by leisuretexpainting** [comment]
Hi Robert, Very sorry, I didn't get these!
At this stage, yes I think that will be the next process -  awaiting take off materials, awaiting quotation, quotation sent.

In regards to the tender to quotation it is a bit complex what we would like the next stage to be. We would actually like 2 different processes - similar to the old system for private and commercial projects, however for the new construction process we would actually like to put in place a csv style document that can upload to the system to fulfill the quotation.

**Comment by rdmordido** [comment]
@leisuretexpainting 

Yes Danielle, I am also thinking of that functionality that will enable our users to upload a csv formatted quotation file then it will be automatically encoded into our own database for a faster and much easier computations and data handling.

I am also having good conversations with our engineer guys here to have a much better idea on how this system will work to help us improve our current process.

I am almost done on initializing a new tender. The remaining task now on which i am working on is handling tender related documents.

**Comment by rdmordido** [comment]
@leisuretexpainting 

Related tasks update for this issue:

On-going:
1. Contractor Management Module
2. Contact Management Module

Modules Completed:

User Modules

a.  Create new user and user roles
b.  Listing page for all created users
c.  Update user details and user roles
d.  Remove existing user

Tender Modules

a.  Create new tender
b.  Auto search existing contractor details
c.  Auto search existing contact details based on selected contractor
d.  Added option to add new contractor and contact details during new tender process
e.  Upload tender related documents
f.  Listing page for all created tenders
g:     Download function for uploaded documents
h:     Update tender detail page
